src/__tests__/App.test.tsx

```diff
@@ -0,0 +1,148 @@
+    expect(wrapper.find('Suspense')).toHaveLength(1)
+    expect(wrapper.find('withI18nextTranslation(Connect(Notification))')).toHaveLength(1)
+    expect(wrapper.find('Switch')).toHaveLength(1)
+    expect(wrapper.find('RouteWithLayout')).toHaveLength(6)
+    expect(wrapper.find('Redirect')).toHaveLength(3)
+  })
+
+  it('should contain specific components if the user is not logged in via shallow', () => {
+    const calculatedProps = {
+      ...props,
+      isAuthenticated: false,
+    }
+
+    const wrapper = shallow(<App {...calculatedProps} />)
+    expect(wrapper.find('ThemeProvider')).toHaveLength(1)
+    expect(wrapper.find('Suspense')).toHaveLength(1)
+    expect(wrapper.find('withI18nextTranslation(Connect(Notification))')).toHaveLength(1)
+    expect(wrapper.find('Switch')).toHaveLength(1)
+    expect(wrapper.find('RouteWithLayout')).toHaveLength(2)
+    expect(wrapper.find('Redirect')).toHaveLength(1)
+  })
+
+  describe('App via mount', () => {
+    it('should call checkAuthentication when mounted', () => {
+      const wrapper = mount(
+        <MemoryRouter>
+          <Provider store={getMockedStore({})}>
+            <App {...props} />
+          </Provider>
+        </MemoryRouter>,
+      )
+      expect(props.checkAuthentication).toHaveBeenCalledTimes(1)
+    })
+
+    it('should contain specific components if user is logged in via mount', () => {
+      const wrapper = mount(
+        <MemoryRouter>
+          <Provider store={getMockedStore({})}>
+            <App {...props} />
+          </Provider>
+        </MemoryRouter>,
+      )
+      expect(wrapper.find(ThemeProvider)).toHaveLength(1)
+      expect(wrapper.find(StylesProvider)).toHaveLength(1)
+      expect(wrapper.find(Notification)).toHaveLength(1)
+      expect(wrapper.find(Switch)).toHaveLength(1)
+      expect(wrapper.find(RouteWithLayout)).toHaveLength(1)
+      expect(wrapper.find(Dashboard)).toHaveLength(1)
+      expect(wrapper.find(Login)).toHaveLength(0)
+      expect(wrapper.find(Logout)).toHaveLength(0)
+    })
+
+    it('should contain specific components if the user is not logged in via mount', () => {
+      const calculatedProps = {
+        ...props,
+        isAuthenticated: false,
+      }
+
+      const wrapper = mount(
+        <MemoryRouter>
+          <Provider store={getMockedStore({})}>
+            <App {...calculatedProps} />
+          </Provider>
+        </MemoryRouter>,
+      )
+      expect(wrapper.find(ThemeProvider)).toHaveLength(1)
+      expect(wrapper.find(StylesProvider)).toHaveLength(1)
+      expect(wrapper.find(Notification)).toHaveLength(1)
+      expect(wrapper.find(Switch)).toHaveLength(1)
+      expect(wrapper.find(RouteWithLayout)).toHaveLength(1)
+      expect(wrapper.find(Dashboard)).toHaveLength(0)
+      expect(wrapper.find(Login)).toHaveLength(1)
+      expect(wrapper.find(Logout)).toHaveLength(0)
+    })
+
+    test('invalid path should redirect to 404', () => {
```

Comment:
what is the difference between test and it ? somewhere test somewhere it

Reply:
Technically there is no difference. I want to say this in the video. There is one small difference that some after **it** expect the word **should**.